internal/remoteproc/remoteproc.go

```diff
@@ -17,10 +18,66 @@ const (
 	rprocFirmwareFileName     = "firmware"
 )
 
-var (
-	rprocFirmwareStorePath = rootpath.Join("lib", "firmware")
-	rprocClassPath         = rootpath.Join("sys", "class", "remoteproc")
-)
+var rprocClassPath = rootpath.Join("sys", "class", "remoteproc")
+
+func getCustomFirmwarePath() (string, error) {
+	customPath, err := os.ReadFile(rootpath.Join("sys", "module", "firmware_class", "parameters", "path"))
```

Comment:
`rootpath.Join("sys", "module", "firmware_class", "parameters", "path")` could be pulled out of this function, similar to `rprocClassPath`.